recursive-feature-elimination/script.whizzml

```diff
@@ -0,0 +1,291 @@
+             evaluations []) 
+          ;; Loop until we obtain the desired number of features
+          (if (<= (count selected) nfeatures)
+              (response selected 
+                        (feature-names fields selected) 
+                        evaluations 
+                        do-evaluation)
+              (let (_ (log-info "------- New iteration -------")
+                    _ (log-info "Creating random forest with " 
+                                (count selected) 
+                                " fields")
+                    ;; Create random forest and obtain its importances
+                    rf-id (random-forest dataset-id obj-id 0 selected)
+                    imps (get-importances (fetch rf-id))
+                    ;; Obtaining minimum field importance
+                    min-imp (minimum-importance imps)
+                    removed-field (filter (lambda (n) (= (imps n) min-imp))
+                                          (keys imps))
+                    ;; Obtain first element, just in case there are two ore 
+                    ;; more fields with the minimum importance
+                    removed-field (removed-field 0)
+                    _ (log-info  "Removed-field " 
+                                 removed-field ": "
+                                 ((feature-names fields [removed-field]) 0))
+                    ;; Remove field from selected list
+                    selected (keys (dissoc imps removed-field))
+                    _ (log-info "There are " 
+                                (count selected) 
+                                " fields after filtering")
+                    ev (when do-evaluation 
+                           (fetch(evaluate-model rf-id test-dataset-id)))
+                    ev-result (when ev (ev ["result" "model" evaluation-metric]))
+                    ;; Iterations evaluation info
+                    iterations (when ev {"features" (count selected)
+                              "evaluation"  ev-result})
+                    evaluations (when ev (append evaluations iterations)))
+              (recur selected
+                      evaluations))))))
+
+;; response
+;;
+;; Creates a human-readable response and clean the created resources
+;;
+;; Inputs:
+;;   selected: (list) Selected fields ids
+;;   selected-names: (list) Selected fields names
+;;   evaluations: (list) Result of the evaluations at each iteration
+;;   do-evaluation: (boolean) Whether we performed evaluations
+;;
+;; Output: (map) Map with selected field names, selected field ids, and 
+;;         iterations evaluation info (only if test-dataset-id is present)
+(define (response selected selected-names evaluations do-evaluation)
+  (let( evaluations (if  do-evaluation
+                         evaluations
+                         ["test-dataset-id needed to create evaluations"]))
+    ; Remove all created resources
+    (for (id (created-resources)) (safe-delete id)) 
```

Comment:
just as a suggestion: (map safe-delete (created-resources)) and could be run before the let block